src/predictor/gpu_predictor.cu

```diff
@@ -230,11 +230,33 @@ class GPUPredictor : public xgboost::Predictor {
       int device = devices_.DeviceId(shard);
       auto data_span = data.DeviceSpan(device);
       dh::safe_cuda(cudaSetDevice(device));
-      // copy the last element from every shard
-      dh::safe_cuda(cudaMemcpy(&offsets.at(shard + 1),
-                               &data_span[data_span.size()-1],
-                               sizeof(size_t), cudaMemcpyDeviceToHost));
+      if (data_span.size() == 0) {
+        offsets[shard + 1] = data.ConstHostVector().back();
```

Comment:
`ConstHostVector()` (and sync to host, in general), is not safe to call in multi-threaded code.

Reply:
Extracted out the value. Would this work?